scripts/osx-setup-opam-for-release.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+set -eux
+
+# Setup the environment under MacOS to build and release semgrep-core.
+
+# history: there used to be a separate osx-m1-release.sh script
+# that was mostly a copy of this file, but now the
+# build steps are identical so we just have one script.
+#
+# Previously this was a combined file with other code to handle forcing static
+# linkage, but this caused issues due to git's runtime dependency on pcre2.
+# Ideally we would have a better way of causing static linking than removing
+# potential runtime deps of other things in the build script.
```

Comment:
```suggestion
# We separate out opam configuration because we have some scripts
which rely
# on building with opam before running the rest of MacOS setup. In particular,
# to link pcre2, we need to install opam, build with it dynamically linked, 
then remove
# it so it can be statically linked.
```
Rewrote this to focus more on the change without context of the other file